driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamOutput.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.changestream;
+
+import com.mongodb.MongoNamespace;
+import org.bson.codecs.Codec;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.codecs.pojo.annotations.BsonCreator;
+import org.bson.codecs.pojo.annotations.BsonId;
+import org.bson.codecs.pojo.annotations.BsonProperty;
+
+/**
+ * Represents the output from a {@code $changeStream} aggregation.
+ *
+ * @param <TDocument> The type that this collection will encode the {@code fullDocument} field into.
+ * @since 3.6
+ */
+public final class ChangeStreamOutput<TDocument> {
+
+    @BsonId()
+    private final ResumeToken resumeToken;
+    @BsonProperty("ns")
+    private final MongoNamespace namespace;
+    private final TDocument fullDocument;
+    private final OperationType operationType;
+    private final UpdateDescription updateDescription;
+
+    /**
+     * Creates a new instance
+     *
+     * @param resumeToken the resume token
+     * @param namespace the namespace
+     * @param fullDocument the fullDocument
+     * @param operationType the operation type
+     * @param updateDescription the update description
+     */
+    @BsonCreator
+    public ChangeStreamOutput(@BsonProperty("resumeToken") final ResumeToken resumeToken,
+                              @BsonProperty("namespace") final MongoNamespace namespace,
```

Comment:
A bit unrelated to the matter at hand, but is it odd that we're overloading the semantics of BsonProperty's value?  The documentation for it says: "The value to use for the mapped property when storing the property in the database".  But here we use it to refer to the property name.  Does Jackson do the same?